Detectors/MUON/MID/Workflow/src/CalibDataProcessorSpec.cxx

```diff
@@ -74,31 +74,39 @@ class FetToDeadDeviceDPL
       }
     }
 
-    std::vector<ColumnData> noisyChannels;
-    noisyChannels.insert(noisyChannels.end(), data[1].begin(), data[1].end());
+    mNoise.clear();
+    mNoiseROF.clear();
+    mDead.clear();
+    mDeadROF.clear();
 
-    std::vector<ROFRecord> noisyChannelsRof;
-    noisyChannelsRof.insert(noisyChannelsRof.end(), dataRof[1].begin(), dataRof[1].end());
+    mNoise.insert(mNoise.end(), data[1].begin(), data[1].end());
+    mNoiseROF.insert(mNoiseROF.end(), dataRof[1].begin(), dataRof[1].end());
 
-    auto deadChannelsPair = getDeadChannels(data[2], dataRof[2], data[0], dataRof[0]);
+    mergeChannels(data[2], dataRof[2], data[0], dataRof[0]);
 
-    pc.outputs().snapshot(of::Output{header::gDataOriginMID, "NOISE", 0}, noisyChannels);
-    pc.outputs().snapshot(of::Output{header::gDataOriginMID, "NOISEROF", 0}, noisyChannelsRof);
-    pc.outputs().snapshot(of::Output{header::gDataOriginMID, "DEAD", 0}, deadChannelsPair.first);
-    pc.outputs().snapshot(of::Output{header::gDataOriginMID, "DEADROF", 0}, deadChannelsPair.second);
+    pc.outputs().snapshot(of::Output{header::gDataOriginMID, "NOISE", 0}, mNoise);
+    pc.outputs().snapshot(of::Output{header::gDataOriginMID, "NOISEROF", 0}, mNoiseROF);
+    pc.outputs().snapshot(of::Output{header::gDataOriginMID, "DEAD", 0}, mDead);
+    pc.outputs().snapshot(of::Output{header::gDataOriginMID, "DEADROF", 0}, mDeadROF);
   }
 
  private:
-  FetToDead mFetToDead{}; ///< FET to dead channels converter
+  FetToDead mFetToDead{};           ///< FET to dead channels converter
+  std::vector<ColumnData> mNoise;   ///< Merged noise
+  std::vector<ROFRecord> mNoiseROF; ///< Merged noise ROFs
+  std::vector<ColumnData> mDead;    ///< Merged dead
+  std::vector<ROFRecord> mDeadROF;  ///< Merged dead ROFs
 
-  std::pair<std::vector<ColumnData>, std::vector<ROFRecord>> getDeadChannels(gsl::span<const ColumnData> fetData, gsl::span<const ROFRecord> fetDataRof, gsl::span<const ColumnData> selfTrigData, gsl::span<const ROFRecord> selfTrigDataRof)
+  void mergeChannels(gsl::span<const ColumnData> fetData, gsl::span<const ROFRecord> fetDataRof, gsl::span<const ColumnData> selfTrigData, gsl::span<const ROFRecord> selfTrigDataRof)
   {
+    // This method selects the self-triggered events that are close in time with a FET event and merges them with the fet event.
+    // This is needed since the detector answer is not perfectly aligned in time.
+    // Since calibration runs occur with no beam, all other self-triggers are actually noise.
+
     int64_t maxDiff = 1;
```

Comment:
Don't you want `minDiff` and `maxDiff` to be either parameters of this `mergeChannel` or data member of the class ? Seems strange to hardcode it within the function ?